app/uk/gov/hmrc/helptosavestub/controllers/NSIController.scala

```diff
@@ -137,6 +138,31 @@ object NSIController extends BaseController with Logging {
         })
   }
 
+  def getTransactions(correlationId: Option[String],
+                      nino:          Option[String],
+                      version:       Option[String],
+                      systemId:      Option[String]): Action[AnyContent] = Action {
+    implicit request ⇒
+      validateParams(correlationId, nino, version, systemId).fold(
+        errors ⇒ {
+          logger.warn("[Handle Transactions Query] invalid params")
+          BadRequest(Json.toJson(NSIErrorResponse(version, correlationId, errors.toList)))
+        }, {
+          validatedNino ⇒
+            if (validatedNino.contains("401")) {
+              Unauthorized
+            } else if (validatedNino.contains("500")) {
+              InternalServerError
+            } else {
+              val maybeAccount = getTransactionsByNino(validatedNino, correlationId)
+              maybeAccount match {
+                case Right(a) ⇒ Ok(Json.toJson(a))
+                case Left(e)  ⇒ BadRequest(Json.toJson(NSIErrorResponse(version, correlationId, Seq(e))))
```

Comment:
Looking at `getTransactionsByNino`, the left case occurs when a NINO is not recognised. Should this not map to a 404 (NOT FOUND)? A way to make the intention of `getTransactionsByNino` more obvious is for it to return an `Option` rather than an `Either` since the error case is always when no account is defined for the NINO.

Reply:
@AndyHWChung I believe the reason we're using `BadRequest` with `Either` to return the error is because we want to mimic the behaviour from atos. 
Because in a NINO not found scenario, we're currently getting the following response from atos: 
400 - Unknown NINO

Reply:
Ah yes, I remember now, thanks for reminding me - we're mapping that case to a 404 in the help-to-save BE microservice, but NS&I return a 400.